src/app/components/add-card/add-card.component.ts

```diff
@@ -81,12 +85,26 @@ export class AddCardComponent implements OnDestroy {
     this.newTask = '';
   }
 
+  onCompleteTask(taskIndex: number){
+    this.taskService.taskList[taskIndex].done = true;
+    this.taskService.taskList[taskIndex].showCompleteButton = false;
+    this.taskService.taskList[taskIndex].showEditButton = false;  
+  }
+
+  calculateDuration(startDate: number): number{
+    this.endDate = Date.now();
+    return Math.floor(Math.abs((this.endDate - startDate)/this.constantsService.MS_PER_DAY) + 1);
+  }
+
   handleTaskButtonClick({id, dataJob}: TaskEventData): void {
     const taskIndex = this.taskService.taskList.findIndex(task => task.id === id);
 
+    if (taskIndex !== -1 && dataJob === this.constantsService.COMPLETE) {
+      this.onCompleteTask(taskIndex);
+    }
     if (taskIndex >= 0 && dataJob === this.constantsService.DELETE_TODO) {
       this.taskService.taskList.splice(taskIndex, 1);
-    }
+    } 
     else{
       return;
     }
```

Comment:
`switch` seems a good alternative for this.

Reply:
done!. please check